apps/aecore/src/aec_parent_connector.erl

```diff
@@ -314,15 +386,82 @@ handle_fetch_block(Fun, Arg,
             Err
     end.
 
-get_network_id() ->
-    gen_server:call(?SERVER, get_network_id).
 
-get_parent_conn_mod() ->
-    gen_server:call(?SERVER, get_parent_conn_mod).
 
-get_sign_module() ->
-    gen_server:call(?SERVER, get_sign_module).
+is_pin(Pin) -> 
+    gen_server:call(?SERVER, {is_pin, Pin}).
+
+% PINREFAC is this (aec_p_c) the correct place for this and the following CC pin related ones?
+-spec get_pinning_data() -> {ok, map()} | {error, atom()}.
+get_pinning_data() ->
+    {ok, #{epoch := Epoch,
+           first := First,
+           last  := Last}} = aec_chain_hc:epoch_info(),
+    lager:debug("Get pin data for epoch ~p for leader of block ~p", [Epoch - 1, Last]),
+    {ok, BlockHash} = aec_chain_state:get_key_block_hash_at_height(First - 1),
+    {ok, ChainType} = get_parent_chain_type(),
+    PrevEpoch = Epoch - 1,
+    Height = First - 1,
+    case aec_consensus_hc:leader_for_height(Last) of
+      {ok, Leader} ->
+        {ok, #{epoch             => PrevEpoch,
+               height            => Height,
+               block_hash        => BlockHash,
+               parent_payload    => encode_parent_pin_payload(#{epoch => PrevEpoch, height => Height, block_hash => BlockHash}),
+               last_leader       => Leader,
+               parent_type       => ChainType,
+               parent_network_id => get_network_id()}};
+      error ->
+          %% schedule not yet cached
+          {error, last_leader_unknown}
+    end.
+
+get_generations(FromHeight, ToHeight) ->
+    lists:flatten(
+        lists:map(
+        fun(Height) ->
+            case aec_chain:get_generation_by_height(Height, backward) of
+                {ok, #{micro_blocks := MBs}} -> MBs;
+                error -> error({failed_to_fetch_generation, Height})
+            end
+        end,
+        lists:seq(FromHeight, ToHeight)
+    )).
+
+
+find_spends_to(Account) ->
+    %% TODO Not happy with this. are we in the correct epoch when this is supposed to be 
```

Comment:
We never need to validate this. The transaction is just there for leader convenience has nothing to do with consensus

Reply:
Correct. If we feel the need to correct this later, we'll fix it then.

Reply:
Another convenience is that these spends enable the middleware to check for pins posted (and possibly validate them as well) to display in timelines etc.